src/main/java/picard/illumina/SortedBasecallsConverter.java

```diff
@@ -199,4 +221,98 @@ private synchronized void addRecord(final String barcode, final CLUSTER_OUTPUT_R
                     IOUtil.filesToPaths(tmpDirs));
         }
     }
+
+    protected void awaitTileProcessingCompletion() throws IOException {
+        tileReadExecutor.shutdown();
+        // Wait for all the read threads to complete before checking for errors
+        ThreadPoolExecutorUtil.awaitThreadPoolTermination("Reading executor", tileReadExecutor, Duration.ofMinutes(5));
+        tileProcessingComplete = true;
+
+        try {
+            // Check for reading errors
+            if (tileReadExecutor.hasError()) {
+                interruptAndShutdownExecutors(tileReadExecutor, completedWorkExecutor, tileWriteExecutor);
+            }
+
+            synchronized (completedWork) {
+                log.debug("Final notification of work complete.");
+                completedWork.notifyAll();
+            }
+
+            // Wait for tile processing synchronization to complete
+            ThreadPoolExecutorUtil.awaitThreadPoolTermination("Tile completion executor", completedWorkExecutor, Duration.ofMinutes(5));
+
+            // Check for tile work synchronization errors
+            if (completedWorkExecutor.hasError()) {
+                interruptAndShutdownExecutors(tileReadExecutor, completedWorkExecutor, tileWriteExecutor);
+            }
+        } finally {
+            // We are all done scheduling work. Now close the writers.
+            for (Writer<CLUSTER_OUTPUT_RECORD> clusterWriter : barcodeRecordWriterMap.values()) {
+                clusterWriter.close();
```

Comment:
I'm not sure if the pool itself is still accessible fro here, but it will be more efficient to call close at the pool level than to loop over each writer. Calling close on the pool actually launches a background task async close the writers instead of calling close, waiting for the writer to drain, and then calling close on the next one.

Reply:
The main issue is that not all basecalls converters will be using the AsyncWriter pool. (like when we convert to sam instead of fastq)